app/src/main/java/nu/yona/app/ui/settings/SettingsFragment.java

```diff
@@ -84,11 +85,11 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
                 }
             }
         });
-        try {
-            PackageInfo pInfo = getActivity().getPackageManager().getPackageInfo(getActivity().getPackageName(), 0);
-            ((TextView) view.findViewById(R.id.label_version)).setText(getString(R.string.version) + pInfo.versionName + getString(R.string.space) + pInfo.versionCode);
-        } catch (PackageManager.NameNotFoundException e) {
-            AppUtils.throwException(SettingsFragment.class.getSimpleName(), e, Thread.currentThread(), null);
+        AppMetaInfo appMetaInfo = AppMetaInfo.getInstance();
+        if(appMetaInfo.getAppVersionCode() != 0){
```

Comment:
What is the purpose of this if-statement?

Reply:
My intention was to avoid crash, if values are not set for reason during build time

Reply:
I'm not in favor of this type of constructs as it complicates the code for no reason. This data comes from the Android metadata, so it's always there. If we ever manage to create a broken build, it should be flagged rather than accepted.

Can you take this construct out and just rely on the metadata?

Reply:
Ok. I will remove it.

Reply:
Can you take this along in your current PR?

Reply:
Done this as part of https://github.com/yonadev/yona-app-android/pull/513